addons/website_livechat/controllers/main.py

```diff
@@ -68,6 +72,14 @@ def get_session(self, channel_id, anonymous_name, previous_operator_id=None, cha
             anonymous_name = _('Visitor #%s') % visitor_sudo.id
         return super(WebsiteLivechat, self).get_session(channel_id, anonymous_name, previous_operator_id=previous_operator_id, chatbot_script_id=chatbot_script_id, persisted=persisted, **kwargs)
 
+    @http.route()
+    def im_livechat_chat_history(self, uuid, last_id=False, limit=20):
+        result = super().im_livechat_chat_history(uuid, last_id=last_id, limit=limit)
+        channel = request.env["discuss.channel"].sudo().search([("uuid", "=", uuid)], limit=1)
+        if channel and not channel.channel_member_ids.filtered(lambda member: member.guest_id):
```

Comment:
could be a partner too

Reply:
only check that if there is no logged in user (if there is a partner, it's already added from `action_send_chat_request`)